packages/react/src/pages/planning-scheduler/use-scheduler-logic.ts

```diff
@@ -0,0 +1,175 @@
+import { useRef, useState, useCallback, useMemo, useEffect } from 'react';
+
+import Tooltip from 'devextreme-react/tooltip';
+import Scheduler from 'devextreme-react/scheduler';
+import { ViewType } from 'devextreme/ui/scheduler';
+import DataSource from 'devextreme/data/data_source';
+
+import { getTasksForScheduler } from 'dx-template-gallery-data';
+import { useScreenSize } from '../../utils/media-query';
+import { findAllAppointmentsForDay } from './utils';
+
+export const useSchedulerLogic = () => {
+  const { isXSmall, isLarge } = useScreenSize();
+  const tooltipRef = useRef<Tooltip>(null);
+  const schedulerRef = useRef<Scheduler>(null);
+
+  const [agendaItems, setAgendaItems] = useState<{ startDate: Date }[]>();
+  const [currentView, setCurrentView] = useState<ViewType>('workWeek');
+  const [date, setDate] = useState(new Date());
+  const [rightPanelOpen, setRightPanelOpen] = useState<boolean>(false);
+  const [selectedAppointment, setSelectedAppointment] = useState<{ data, target }>();
+  const [tasks, setTasks] = useState<DataSource>();
+
+  useEffect(() => {
+    getTasksForScheduler().then(tasksList => {
+      setTasks(new DataSource(tasksList));
```

Comment:
Since setTasks is called only in this effect, you can merge both effects into one and call setAgendaItems after receiving the taskList.

Reply:
It seemed like a good suggestion, bit as it figured out, if we create new Datasource and immediately
after that call DataSource.items() it will return empty array, which results as empty agenda in right sight panel
Seems like DataSource asynchronously sets it's items, therefore i'll bring back second useeffect